src/powershell-native/nativemsh/pwrshcommon/CMakeLists.txt

```diff
@@ -10,6 +10,18 @@ add_library(pwrshcommon
 
 target_include_directories(pwrshcommon PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 
+#
+# VS 2017 corrupts the INCLUDE environment variable when processing the csproj
+# file generated for this template. The netfxsdk path is shortened from
+# C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\Include\um
+# to
+# Include\um
+# This prevents msbuild from locating corerror.h and throws a resolution error
+# during compilation. We work around the problem by force-including the
+# contents of the environment variable before it gets corrupted.
+#
+target_include_directories(pwrshcommon PUBLIC $ENV{INCLUDE})
```

Comment:
That is the scenario where I ran it. My system has VS 2015 and VS 2017 installed side-by-side. I have not verified it on a system that only has VS 2017. As long as vcvarsall.bat generates the correct INCLUDE environment variable, it should work though.